ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -555,11 +559,22 @@ peerSelectionGovernorLoop tracer
       -- get the current time after the governor returned from the blocking
       -- 'evalGuardedDecisions' call.
       now <- getMonotonicTime
+
       let Decision { decisionTrace, decisionJobs, decisionState } =
             timedDecision now
           !newCounters = peerStateToCounters decisionState
 
+          !outboundConnectionsState =
+            if activePeers st
+               `Set.isSubsetOf`
+               LocalRootPeers.keysSet (localRootPeers st)
+            then ConnectedToOnlyLocalOutboundPeers
+            else ConnectedToExternalOutboundPeers
+
       atomically $ do
+        -- Update outbound connections state
+        updateOutboundConnectionsState actions outboundConnectionsState
+
```

Comment:
I think this logic may cause trouble.
We don't check for trusted peers here, is is possible that we could respond with `ConnectedToExternalOutboundPeers` before we even managed to disconnect from all untrusted peers when entering the TooOld state?